server/src/main/java/org/apache/druid/server/coordinator/CuratorLoadQueuePeon.java

```diff
@@ -181,10 +183,10 @@ public int getNumberOfSegmentsInQueue()
   }
 
   @Override
-  public void loadSegment(final DataSegment segment, @Nullable final LoadPeonCallback callback)
+  public void loadSegment(final DataSegment segment, SegmentAction action, @Nullable final LoadPeonCallback callback)
   {
-    SegmentHolder segmentHolder = new SegmentHolder(segment, Action.LOAD, Collections.singletonList(callback));
-    final SegmentHolder existingHolder = segmentsToLoad.putIfAbsent(segment, segmentHolder);
+    QueuedSegment segmentHolder = new QueuedSegment(segment, action, callback);
```

Comment:
Yes, I will rename `QueuedSegment` back to `SegmentHolder` as it is adding unnecessarily to the diff clutter.

Reply:
Reverted to use `SegmentHolder`.

Reply:
> That said, I would've expected the queue itself to return a QueuedSegment rather than something creating one of those before it's actually queued...

Do you mean like a `computeIfAbsent` rather than `putIfAbsent`?